.eslintrc.js

```diff
@@ -0,0 +1,52 @@
+module.exports = {
+	root: true,
+
+	rules: {
+		// automatically fixable stylistic rules
+		'arrow-spacing': 2,
+		'comma-dangle': [2, 'never'],
+		'dot-notation': 2,
+		'eol-last': [2, 'never'],
+		indent: [2, 'tab', { SwitchCase: 1 }],
+		'keyword-spacing': [2, { before: true, after: true }],
+		'no-trailing-spaces': 2,
+		'no-var': 2,
+		'object-shorthand': [2, 'always'],
+		'one-var': [2, 'never'],
+		'prefer-arrow-callback': 2,
+		'prefer-const': [2, { destructuring: 'all' }],
+		'quote-props': [2, 'as-needed'],
+		semi: [2, 'always'],
+		'space-before-blocks': [2, 'always'],
+
+		// eslint:recommended overrides
+		'no-cond-assign': 0,
+		'no-inner-declarations': 0,
+		'no-mixed-spaces-and-tabs': [2, 'smart-tabs']
+	},
+
+	plugins: ['@typescript-eslint'],
+
+	extends: [
+		'eslint:recommended',
+		'plugin:import/errors',
+		'plugin:import/warnings',
+		'plugin:import/typescript'
+		// TODO: fix typerscript files
+		// 'plugin:@typescript-eslint/recommended',
+	],
+
+	parser: '@typescript-eslint/parser',
+	parserOptions: {
+		ecmaVersion: 9,
```

Comment:
As far as I'm aware, `ecmaVersion` and `sourceType` are not required to [configure](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#configuration) `@typescript-eslint/parser`. It should pick up the relevant settings from `tsconfig.json`.

I've also found setting `tsconfigRootDir: __dirname,` to be useful in some scenarios like VS Code extensions.

Reply:
I thought so too, but it complains about `Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context` for some reason.

Reply:
Had a deeper dive into my own config and found that even though I leave these values out, a preset I extend from is actually adding it (`eslint-config-airbnb-base`). Sorry, I didn't realise. You're right in that it throws the error!

Learnt something new :)